bottlerocket/agents/src/sonobuoy.rs

```diff
@@ -123,72 +170,43 @@ pub fn results_sonobuoy(
-        .as_array()
-        .context(error::MissingSonobuoyStatusFieldSnafu { field: "plugins" })?
-        .first()
-        .context(error::MissingSonobuoyStatusFieldSnafu {
-            field: format!("plugins.{}", sonobuoy_config.plugin),
-        })?;
-
-    // Sometimes a helpful log is available in the progress field, but not always.
-    let progress_status = e2e_status
-        .get("progress")
-        .map(|value| value.to_string())
-        .unwrap_or_else(|| "".to_string());
-
-    let result_status = e2e_status
-        .get("result-status")
-        .context(error::MissingSonobuoyStatusFieldSnafu {
-            field: format!("plugins.{}.result-status", sonobuoy_config.plugin),
-        })?
-        .as_str()
-        .context(error::MissingSonobuoyStatusFieldSnafu {
-            field: format!("plugins.{}.result-status", sonobuoy_config.plugin),
-        })?;
-
-    let result_counts = run_status
+    let (passed, failed, skipped) = run_status
         .get("plugins")
         .context(error::MissingSonobuoyStatusFieldSnafu { field: "plugins" })?
         .as_array()
         .context(error::MissingSonobuoyStatusFieldSnafu { field: "plugins" })?
-        .first()
-        .context(error::MissingSonobuoyStatusFieldSnafu {
-            field: format!("plugins.{}", sonobuoy_config.plugin),
-        })?
-        .get("result-counts")
-        .context(error::MissingSonobuoyStatusFieldSnafu {
-            field: format!("plugins.{}.result-counts", sonobuoy_config.plugin),
-        })?;
-
-    let num_passed = result_counts
-        .get("passed")
-        .map(|v| v.as_u64().unwrap_or(0))
-        .unwrap_or(0);
-
-    let num_failed = result_counts
-        .get("failed")
-        .map(|v| v.as_u64().unwrap_or(0))
-        .unwrap_or(0);
-
-    let num_skipped = result_counts
-        .get("skipped")
-        .map(|v| v.as_u64().unwrap_or(0))
-        .unwrap_or(0);
+        .iter()
+        .fold((0, 0, 0), |acc, x| {
+            let (pass, fail, skip) = if let Some(result_counts) = x.get("result-counts") {
+                let pass = result_counts
+                    .get("passed")
+                    .and_then(|count| count.as_u64())
+                    .unwrap_or(0);
+                let fail = result_counts
+                    .get("failed")
+                    .and_then(|count| count.as_u64())
+                    .unwrap_or(0);
+                let skip = result_counts
+                    .get("skipped")
+                    .and_then(|count| count.as_u64())
+                    .unwrap_or(0);
+                (pass, fail, skip)
+            } else {
+                (0, 0, 0)
+            };
+            (acc.0 + pass, acc.1 + fail, acc.2 + skip)
+        });
 
     Ok(TestResults {
-        outcome: match result_status {
-            "pass" | "passed" => Outcome::Pass,
-            "fail" | "failed" => Outcome::Fail,
-            "timeout" | "timed-out" => Outcome::Timeout,
```

Comment:
Have we lost something by not checking for timeouts? The main thing I'm worried about is that there is that sometimes e2e "runs" for a long time but nothing happens. At the end I think it reports no tests have run. I'm not sure how we were handing that before and whether this changes it.